.github/workflows/infrastructure.yaml

```
@@ -22,26 +21,22 @@ permissions:
   pull-requests: write
 
 jobs:
-  terraform:
+  plan:
+    if: github.event_name == 'pull_request'
     strategy:
       fail-fast: false
       matrix:
-        config:
-          - environment: dev
-            account_id: <DEV_ACCOUNT_ID>
-            tf_deploy_override: ${{ github.event_name == 'workflow_dispatch' && true || false }} # Allow manual deployment of Dev
-          # - environment: prod
-          #   account_id: <PROD_ACCOUNT_ID>
-          #   tf_deploy_override: false
+        environment: <MATRIX>
+    uses: tx-pts-dai/github-workflows/.github/workflows/tf-plan.yaml@v0.16.0
+    with:
+      environment: ${{ matrix.environment }}
 
-    uses: tx-pts-dai/github-workflows/.github/workflows/tf-deploy-basic.yaml@v0.8.0
+  apply:
+    if: github.ref_name == github.event.repository.default_branch && (github.event_name == 'push' || github.event_name == 'workflow_dispatch')
+    strategy:
+      fail-fast: false
+      matrix:
+        environment: $MATRIX
```

You meant `<MATRIX>` right?
```suggestion
        environment: <MATRIX>
```